### PR TITLE
Update `opentelemetry-user_events-*` and `opentelemetry-etw-*` exporters to use otel v0.24

### PR DESCRIPTION
**Pull request by lalitb**
Fixes #
- Update `opentelemetry-user_events-*` and `opentelemetry-etw-*` exporters to use otel v0.24
- `opentelemetry-datadog` and `opentelemetry-stackdriver` continue using otel v0.23. Need help from the users/contributors for these repos to do the upgrade.
- `opentelemetry-zpages` and `opentelemetry-aws` don't need any significant change for the new versions, so they are implicitly bumped.

## Changes

Please provide a brief description of the changes here.

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust-contrib/blob/main/CONTRIBUTING.md) guidelines followed
* [x] Unit tests added/updated (if applicable)
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [x] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-aws/Cargo.toml [hunk 1]
```diff
@@ -31,5 +31,5 @@ opentelemetry_sdk = { workspace = true, optional = true }
 opentelemetry_sdk = { workspace = true, features = ["testing"] }
```
Comment (cijothomas) [inline opentelemetry-aws/Cargo.toml:31]: given the update to these crates, we need a changelog entry. or force them to use 0.23 itself

Comment (lalitb) [reply opentelemetry-aws/Cargo.toml:31]: Oh yes, there were few more which actually don't need any change. Updated their changelog.

File: opentelemetry-zpages/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,16 @@
 # Changelog
 
-## vNext
+## Unreleased
+
+### Changed
```
Comment (cijothomas) [inline opentelemetry-zpages/CHANGELOG.md:5]: 0.7.0 is repeated. Also both api and sdk versions are bumped?

**Review by cijothomas — APPROVED** [review]
LGTM.
Have left a qn about Zpages changes that should be fixed before merge.

File: opentelemetry-zpages/CHANGELOG.md (hunk 2 above)
Comment (lalitb) [reply opentelemetry-zpages/CHANGELOG.md:5]: Good catch. Fixed both.

**Pull request merged**